### ISSUE TITLE
Unknown error regarding Isaac Gym when configuring GLEAM and running it on Ubuntu 22.04

### ISSUE THREAD

**Issue by jiangyimiao** [zjwzcx/GLEAM#14]
Hi, thank you for your work of GLEAM. I have encountered a question that may be related to environment configuration and want to ask:
I installed all the environments according to the instructions, except for my **Ubuntu version is 22.04**. Although Isaac Gym no longer supports 22.04, I can still run its examples.
However, when running any training or evaluation scripts provided, the program will get stuck in the **self._create_envs()** statement of the **create_sim()** of **class DroneRobot (BaseTask)** in the **drone_robot.py** file, meaning that after running **self. _create_terrain()**, it will not enter **self. _create_envs()**.

I have added some debugging print information as follows:

```
def create_sim(self):
        """ Creates simulation, terrain and evironments
        """
        print("===  Start creating simulation environment ===")
        self.up_axis_idx = 2  # 2 for z, 1 for y -> adapt gravity accordingly
        print("=== Call gym.creat_Sim ===")
        self.sim = self.gym.create_sim(
            self.sim_device_id, self.graphics_device_id, self.physics_engine, self.sim_params
        )
        print("=== Gym.creat_Sim completed ===")
        print("=== Start creating terrain ===")
        self._create_terrain()
        print("=== Terrain creation completed ===")
        self._create_envs()
        print("=== Simulated environment created successfully ===")
```

```
def _create_envs(self):
        """ Creates environments:
             1. loads the robot URDF/MJCF asset,
             2. For each environment
                2.1 creates the environment, 
                2.2 calls DOF and Rigid shape properties callbacks,
                2.3 create actor with these properties and add them to the env
             3. Store indices of different bodies of the robot
        """
        print("=== Start creating environment (_create_envs) ===")
        
    
        asset_path = self.cfg.asset.file.format(LEGGED_GYM_ROOT_DIR=OPEN_ROBOT_ROOT_DIR)

    
        asset_root = os.path.dirname(asset_path)
        asset_file = os.path.basename(asset_path)

        print(f"=== Resource Path: {asset_path} ===")
        ……
```

The return information when running any training or evaluation script is as follows:

> Importing module 'gym_38' (/home/mine/software/isaacgym/python/isaacgym/_bindings/linux-x86_64/gym_38.so)
> Setting GYM_USD_PLUG_INFO_PATH to /home/mine/software/isaacgym/python/isaacgym/_bindings/linux-x86_64/usd/plugInfo.json
> PyTorch version 2.0.0+cu118
> Device count 6
> /home/mine/software/isaacgym/python/isaacgym/_bindings/src/gymtorch
> Using /home/mine/.cache/torch_extensions/py38_cu118 as PyTorch extensions root...
> Emitting ninja build file /home/mine/.cache/torch_extensions/py38_cu118/gymtorch/build.ninja...
> Building extension module gymtorch...
> Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
> ninja: no work to do.
> Loading extension module gymtorch...
> 2025-11-17 15:15:40
> [LOGGING] We start logging training data into /home/mine/Project/GLEAM/runs/train_gleam_stage1_1117_1515_40
> Setting seed: 1
> ===  Start creating simulation environment ===
> === Call gym.creat_Sim ===
> Not connected to PVD
> +++ Using GPU PhysX
> Physics Engine: PhysX
> Physics Device: cuda:2
> GPU Pipeline: enabled
> === Gym.creat_Sim completed ===
> === Start creating terrain ===
> === Terrain creation completed ===
> [Error] [carb.gym.plugin] cudaImportExternalMemory failed on rgbImage buffer with error 999

Using breakpoint debugging, it was found that the program cannot reach **def _create_envs(self)** after passing through the breakpoint **self._create_envs()** breakpoint. I speculate that the problem may occur in the internal call of Isaac Gym. Since I have checked the installation integrity of Isaac Gym and can run the example, I speculate that there may be some conflicts caused by the version of Ubuntu 22.04.

**I would like to know if you have tried configuring GLEAM on Ubuntu 22.04 or encountered similar issues. I would greatly appreciate it if you could provide any reference suggestions!**

**Comment by zjwzcx** [comment]
> Thank you for your reply! I am configuring GLEAM on a server consisting of several 4090, so I cannot change the Ubuntu version at will, and there are currently no other PCs that can meet the configuration requirements. In addition, I need to add an issue that **multiple cards are occupied** when I run the script, even if I specify the card (eg: --sim_device=cuda:0) in the command. For example, when running a command, checking nvidia-smi returns the following message:
> 
> ```
> +------------------------------------------------------------------------------------------+
> | Processes:                                                                                                                        |
> |  GPU   GI   CI              PID   Type   Process name                        GPU Memory  |
> |            ID   ID                                                               Usage   |
> |==========================================================================================|
> |    0   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> |    0   N/A  N/A          864411    C+G   ...aconda3/envs/gleam/bin/python         500MiB |
> |    1   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> |    2   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> |    2   N/A  N/A          864411      C   ...aconda3/envs/gleam/bin/python        2342MiB |
> |    3   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> |    3   N/A  N/A          864411      G   ...aconda3/envs/gleam/bin/python           6MiB |
> |    4   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> |    5   N/A  N/A            2462      G   /usr/lib/xorg/Xorg                         4MiB |
> +------------------------------------------------------------------------------------------+
> ```
> 
> I suspect that the problem I encountered is strongly related to the issue of multiple cards usage, as I have not found any relevant code for calling multiple cards. Therefore, I would like to ask **if there is any setting in GLEAM's code to actively call multiple cards, or if similar problems have been encountered before**? In addition, I would like to ask if you are considering creating a **Docker image for GLEAM**. I believe that having a complete image that can be used would greatly enhance the impact of GLEAM, as I do think it is a great job. However, I am troubled by the inability to reproduce it smoothly due to the inability to change the Ubuntu version. Thank you again for your reply!

GPU0 memory occupancy issue: You’re right—this appears to be a known quirk (a bug or a feature? I don't know..) in Isaac Gym itself. From what I’ve observed and based on my labmates’ experiences, this issue also arises when working with other codebases that use Isaac Gym. To my knowledge, it has no negative impact on runtime or performance, so feel free to ignore it!

Your suggestion about Docker support is really valuable. I’ll consider integrating these improvements into future iteration. Thank you for your attention and suggestion :)
